crates/red_knot_python_semantic/resources/mdtest/comparison/instances/membership_test.md

```diff
@@ -0,0 +1,125 @@
+# Comparison: Membership Test
+
+In Python, "membership test operators" refer to `in` and `not in` operator. To customize their behavior, classes can implement methods like `__contains__`, `__iter__`, or `__getitem__`.
+
+For references, see:
+
+- <https://docs.python.org/3/reference/expressions.html#membership-test-details>
+- <https://docs.python.org/3/reference/datamodel.html#object.__contains__>
+
+## Implements `__contains__`
+
+Classes can support membership tests by implementing the `__contains__` method:
+
+```py
+class A:
+    def __contains__(self, item: str) -> bool: ...
```

Comment:
Thank you so much for taking the time to make all these detailed adjustments—it must have been quite a task!

I have a follow-up question regarding the points you mentioned. Should we also add `from __future__ import annotations`?

```py
from __future__ import annotations  #required?!

class A:
    def foo(self, o: A) -> A:
        return A()
```

Reply:
Yes, we should probably add `from __future__ import annotations` where we are relying on these self-annotations. We actually understand them even without it, because we don't currently model the fact that the class body scope runs immediately at class definition time, but probably we should model that correctly in future.